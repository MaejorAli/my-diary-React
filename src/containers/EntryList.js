import React from 'react';

const dateType = (date) => {
  const parsedDate = new Date(date);
  const months = ['January', 'February', 'March', 'April', 'May', 'June', 'July', 'August', 'September', 'October', 'November', 'December'];
  const entryDate = `${months[parsedDate.getMonth()]} ${parsedDate.getDate()}, ${parsedDate.getFullYear()}`;
  return entryDate;
};

const EntryList = (props) => (
      <>
        <div className="entry-description" id="entries">
          <ul>
            <li>
              <h2>{props.title}</h2>
            </li>

            <li>
              <p>{dateType(props.createdat)}</p>
            </li>
            <li>
              <button type="button" id="entries" onClick={() => props.history.push(`/entries/${props.id}`)}>Read More</button>
            </li>
          </ul>
        </div>
      </>
);

export default EntryList;
