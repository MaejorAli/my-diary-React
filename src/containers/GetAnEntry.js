import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import swal from 'sweetalert';
import { connect } from 'react-redux';
import { getAnEntry } from '../actions/entryActions';
import deleteEntry from '../actions/deleteEntry';


export class GetAnEntry extends Component {
  constructor() {
    super();
    this.deleteEntry = this.deleteEntry.bind(this);
    this.state = {
      entry: {},
      title: '',
    };
  }

  componentDidUpdate(prevProps) {
    const { deleteEntryReducer, entry, history } = this.props;
    if (prevProps.deleteEntryReducer.success !== deleteEntryReducer.success && deleteEntryReducer.success) {
      swal('Good job!', 'Entry Saved Successfully!', 'success');
      setTimeout(() => history.push('/entries'), 3000);
    } else if (prevProps.deleteEntryReducer.error !== deleteEntryReducer.error && deleteEntryReducer.error) {
      swal('Error!', 'Something Went Wrong!', 'error');
    }
    if (prevProps.entry !== entry) this.setState({ title: entry.title });
  }

  componentDidMount() {
    const { match, getAnEntry: getEntry } = this.props;
    const { entryId } = match.params;

    if (entryId) {
      getEntry(entryId);
    }
  }


  static getDerivedStateFromProps(props) {
    return {
      entry: props.entry,
      title: props.entry.title,
    };
  }


  deleteEntry() {
    this.props.deleteEntry(this.state.entry.id);
  }

  render() {
    const { entry, history } = this.props;
    const { title, content } = entry;

    return (
      <div className="get-entry-body">
        <header>
          <div id="menu">
            <Link to="/" className="menuitem">Home</Link>
            <Link to="/" className="menuitem">Profile</Link>
            <Link to="/" className="menuitem">New Entry</Link>
            <Link to="/" className="menuitem">Sign Out</Link>

          </div>
        </header>

        <section>
          <h6 id="date">{new Date().toDateString()}</h6>


          <article>
            <div id="errors" />
            <h1 id="title">{this.state.title}</h1>
            <p id="content">{content}
            </p>

          </article>
        </section>
        <button type="button" id="modifyButton" onClick={() => history.push(`/entries/update/${entry.id}`)}>Modify</button> <button type="button" id="deleteButton" onClick={this.deleteEntry}>Delete</button>


        <footer>
          <div className="container">
         &copy; My Diary 2018
          </div>
        </footer>
      </div>
    );
  }
}
const mapStateToProps = (state) => ({
  entry: state.entries.entry,
  deleteEntryReducer: state.deleteEntryReducer.status,
});
export default connect(mapStateToProps, {
  getAnEntry, deleteEntry,
})(GetAnEntry);
