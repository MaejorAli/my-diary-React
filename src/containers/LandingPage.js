import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import swal from 'sweetalert';
import { signup } from '../actions/authActions';
import '../styles/views/LandingPage.scss';

const initialState = {
  firstname: '',
  lastname: '',
  email: '',
  password: '',
  confirmPassword: '',
};

export class LandingPage extends Component {
 state = initialState;

   handleSubmit = (event) => {
     event.preventDefault();
     const { signup: signupUser, history } = this.props;
     const {
       firstname, lastname, email, password, confirmPassword,
     } = this.state;


     const body = {
       firstname: firstname.trim(),
       lastname: lastname.trim(),
       email: email.trim(),
       password: password.trim(),
     };
     if (confirmPassword !== password) {
       swal('Passwords do not Match!', 'Please check your password', 'warning');
     } else {
       signupUser(body, () => history.push('/user/profile'));
       this.setState(initialState);
     }
   }

   render() {
     const handleChange = (text, key) => this.setState({ [key]: text });
     const {
       firstname, lastname, email, password, confirmPassword,
     } = this.state;
     return (
       <div className="landing-page-body">
         <header>
           <div id="menu">
             <Link to="/" className="menuitem">Home</Link>
             <Link to="/" className="menuitem">Contact Us</Link>
             <Link to="/" className="menuitem">Sign In</Link>
           </div>
         </header>
         <section>
           <div className="content">
             <h1>Welcome to My Diary</h1>
             <p>Simple, classy journal just for you</p>

           </div>

           <article>
             <div className="card card-container">
               <p id="profile-name" className="profile-name-card" />
               <form className="form-signin">
                 <span id="reauth-email" className="reauth-email" />
                 <input type="firstname" value={firstname} id="firstname" className="form form-control" placeholder="First name" onChange={(e) => handleChange(e.target.value, 'firstname')} required />
                 <input type="firstname" value={lastname} id="lastname" className="form form-control" placeholder="Last name" onChange={(e) => handleChange(e.target.value, 'lastname')} required />
                 <input type="email" id="email" value={email} className="form-control" placeholder="Email address" onChange={(e) => handleChange(e.target.value, 'email')} required />
                 <input type="password" id="password" value={password} className="form-control" placeholder="Password" onChange={(e) => handleChange(e.target.value, 'password')} required />
                 <input type="password" id="confirm-password" value={confirmPassword} className="form-control" placeholder="Confirm Password" onChange={(e) => handleChange(e.target.value, 'confirmPassword')} required />
                 <button className="btn btn-lg btn-primary btn-block btn-signin" type="submit" id="enter-signup" onClick={this.handleSubmit}>Sign Up</button>
                 <Link to="/" className="forgot-password">
                  Already Registered?
                 </Link>
               </form>
               <div id="errors" />
             </div>
           </article>
         </section>
       </div>
     );
   }
}

const mapStateToProps = (state) => ({ signupUser: state.auth });

export default connect(mapStateToProps, { signup })(LandingPage);
