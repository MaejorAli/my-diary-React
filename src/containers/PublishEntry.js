import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import swal from 'sweetalert';
import publishEntry from '../actions/publishEntry';

export class PublishEntry extends Component {
  state = {
    title: '',
    content: '',
  }

  componentDidUpdate(prevProps) {
    const { status } = this.props;
    if (prevProps.status.success !== status.success && status.success) {
      swal('Good job!', 'Article Saved Successfully!', 'success');
    } else if (prevProps.status.error !== status.error && status.error) {
      swal('Error!', 'Something Went Wrong!', 'error');
    }
  }


  handleSubmit = (event) => {
    event.preventDefault();
    const { title, content } = this.state;
    if (!title) return swal('Please add a title', 'Your entry needs to have a title', 'warning');
    if (!content) { return swal('Please add a content', 'The content of your article cannot be empty!', 'warning'); }
    const entryPayload = {
      ...this.state,
    };
    // eslint-disable-next-line react/destructuring-assignment
    this.props.publishEntry(entryPayload);
  }

  render() {
    const handleChange = (text, key) => this.setState({ [key]: text });
    const { title, content } = this.state;
    const { error } = this.props;
    if (error) {
      return (
        <div className="no-record centralizer">
          <span>{error}</span>
        </div>
      );
    }
    return (
      <div className="entry-body">
        <header>
          <div id="menu">
            <Link to="/" className="menuitem">Home</Link>
            <Link to="/" className="menuitem">Entries</Link>
            <Link to="/" className="menuitem">Profile</Link>
            <Link to="/" className="menuitem"> Sign Out</Link>
          </div>
        </header>

        <section>
          <div className="date">
            <h5 id="date">{new Date().toDateString()}</h5>
          </div>


          <article>
            <form>
              <div id="errors" />
              <input type="title" id="title" value={title} className="form form-control" placeholder="Title" onChange={(e) => handleChange(e.target.value, 'title')} />
              <textarea name="message" id="content" value={content} placeholder="Write your story..." onChange={(e) => handleChange(e.target.value, 'content')} />
              <button type="button" id="publishEntry" onClick={this.handleSubmit}>Publish</button>
            </form>

          </article>
        </section>


        <footer>
          <div className="container">
     &copy; My Diary 2018
          </div>
        </footer>
      </div>
    );
  }
}
const mapStateToProps = ({
  publishEntryReducer: { status, publishedEntry, error },
}) => ({
  status,
  publishedEntry,
  error,
});
export default connect(mapStateToProps, { publishEntry })(PublishEntry);
