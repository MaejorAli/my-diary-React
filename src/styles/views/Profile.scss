.profile-body{
  font-family: 'Roboto', sans-serif;
  font-weight: 300;
  margin: 0px;
  #errors{
    color: red;
    }
    
    
    #menu{
    text-align: center;
      font-weight: bold;
      
    }
    .menuitem{
    padding: 10px;
      cursor: pointer;
      text-decoration: none;
      color: white;
      font-weight: 300;
    }
    .menuitem:hover{
    background-color: transparent;
      color: grey;
      
    }
    * {
      box-sizing: border-box;
    }
    
    
    
    /* Create two columns/boxes that floats next to each other */
    .part {
    float: left;
    width: 15%;
    color: #96858F;
    padding: 20px;
    }
    
    
    article {
    float: left;
    padding: 20px;
    width: 80%;
    background:  #f1f1f1;
    margin:20px;
    }
    
    /* Clear floats after the columns */
    section:after {
    content: "";
    display: table;
    clear: both;
    }
    
    
    /* Responsive layout - makes the two columns/boxes stack on top of each other instead of next to each other, on small screens */
    @media (max-width: 600px) {
    .part, article {
        width: 100%;
        height: auto;
    }
    }
    
    
    /* Style the header */
    header {
      
    padding: 30px;
    text-align: center;
    background-color: #9099A2;
    }
    
    footer{
      background-color: #F1F1F1;
      text-align: center;
      padding: 10px;
      text-align: center;
    }
    [class*="entypo-"]:before {
      font-family: 'entypo', sans-serif;
    }
    
    
    
    .user-profile {
      margin: auto;
      width: 25em; 
      height: 11em;
      background: #fff;
      border-radius: .3em;
    }
    
    .user-profile  .username {
      margin: auto;
      margin-top: -4.40em;
      margin-left: 5.80em;
      color: gray;
      font-size: 1.53em;
      font-family: "Coda", sans-serif;
      font-weight: bold;
      padding: 10px
    }
    .user-profile  .email {
      margin: auto;
      display: inline-block;
      margin-left: 10.43em;
      color: gray; 
      font-size: .87em;
      font-family: "varela round", sans-serif;
    }
    .user-profile > img.avatar {
      padding: 1em;
      margin-left: .3em;
      margin-top: .3em;
      height: 10em;
      width: 10em;
      border-radius: 18em;
    }
    .user-profile > h4 {
      margin: auto;
      margin-top: 1.35em;
      margin-right: 4.43em;
      width: 14em;
      color: #96858F;
      font-size: .87em;
      font-family: "varela round", sans-serif;
    }
    
    
    
    h2 {
      color: black;
      margin: 0;
      text-align: left;
    }
    
    h1{
      color: grey;
    }
    
    p {
      color: grey;
      margin: 0;
      text-align: left;
    
    }
    
    .entry {
      margin: 60px auto;
      padding: 10px 0px 0px 0px;
      width: 90%;
    }
    
    .entry-img {
      display: inline-block;
      margin-top: 20px;
    }
    .index:link{
        text-decoration: none;
    }
    .index{
    color: black;
      font-family: 'Roboto', sans-serif;
    }
    .entry-description {
      display: inline-block;
      width: 70%;
    }
    
    ul {
      list-style: none;
     }
     input[type=submit] {
      text-align: center;
      padding: 8px 10px;
      font-size: 16px;
      text-align: center;
      cursor: pointer;
      outline: none;
      color: #fff;
      background-color: #96858F;
      border: none;
      border-radius: 15px;
    }
    
    input[type=submit]:hover {background-color: #3e8e41}
    
    input[type=submit]:active {
    background-color: #96858F;
     box-shadow: 0 3px #666;
     transform: translateY(4px);
    }
    input[type=radio]{
     color: #96858F;
    }
    
    button {
    text-align: center;
    padding: 8px 10px;
    font-size: 16px;
    text-align: center;
    cursor: pointer;
    outline: none;
    color: #fff;
    background-color: #96858F;
    border: none;
    border-radius: 15px;
    }
    
    button:hover {background-color: #9099A2;}
    
    button:active {
    background-color: #9099A2;
    box-shadow: 0 3px #666;
    transform: translateY(4px);
    }
    .image-upload > input
    {
      display: none;
    }  
} 
