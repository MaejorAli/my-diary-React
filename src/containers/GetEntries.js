import React, { Component } from 'react';
import { Link } from 'react-router-dom';
// import swal from 'sweetalert';
import { connect } from 'react-redux';
import EntryList from './EntryList';
import { getEntries } from '../actions/entryActions';


export class GetEntries extends Component {
  componentDidMount() {
    const { getAllEntries } = this.props;
    getAllEntries();
  }


  render() {
    const { entries } = this.props;
    // const listEntries = entries.map((entry) => {
    //  <h2>{entry.title}</h2>;
    // });

    // const { title } = entries;
    return (
      <div className="entry-body">
        <header>

          <div id="menu">
            <Link to="/" className="menuitem">Home</Link>
            <Link to="/" className="menuitem">Profile</Link>
            <Link to="/" className="menuitem">New Entry</Link>
            <Link to="/" className="menuitem">Sign Out</Link>
          </div>
        </header>
        <h1>My Stories</h1>
        <div id="errors" />
        <div className="entry">
          {
            entries.map((entry) => <EntryList key={entry.id} {...entry} {...this.props} />)
          }

        </div>

        <footer>
          <div className="container">
     &copy; My Diary 2018
          </div>
        </footer>
      </div>
    );
  }
}
const mapStateToProps = (state) => ({
  entries: state.entries.entries,
});

export default connect(mapStateToProps,
  { getAllEntries: () => getEntries() })(GetEntries);
