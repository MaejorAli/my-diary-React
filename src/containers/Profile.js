import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { getEntries } from '../actions/entryActions';
import updateProfileImage from '../actions/updateProfileImage';
import profilephoto from '../assets/profilephoto.png';
import getProfile from '../actions/profile';
import EntryList from './EntryList';

export class Profile extends Component {
state = {
  image: '',
}

  componentDidMount = () => {
    const { getUserProfile, getAllEntries } = this.props;
    getUserProfile();
    getAllEntries();
  }

  handleSubmit = (e) => {
    const file = e.target.files[0];
    const image = new FormData();
    image.append('image', file);
    // eslint-disable-next-line react/destructuring-assignment
    this.props.updateImage(image);
  }

  render() {
    const { gottenProfile, entries } = this.props;
    return (
      <div className="profile-body">
        <header>
          <div id="menu">
            <Link to="/dashboard" className="menuitem">Home</Link>
            <Link to="/entries" className="menuitem">Entries</Link>
            <Link to="/entries/create" className="menuitem">New Entry</Link>
            <Link to="/" className="menuitem">Sign Out</Link>
          </div>
        </header>

        <div className="user-profile">
          <img className="avatar" id="image" src={gottenProfile.userimage || profilephoto} alt="Ali" />
          <div className="username" id="username">{gottenProfile.firstname} {gottenProfile.lastname}</div>
          <div className="email" id="email">{gottenProfile.email}</div>
          <h4>Published Stories <span id="entryCount" />{entries.length}</h4>
        </div>
        <section>
          <div className="part">
            <div className="image-upload">
              <label htmlFor="file-input">
                <h5>Change Profile Picture</h5>
              </label>
              <input id="file-input" type="file" onChange={this.handleSubmit} />
            </div>
          </div>
          <div id="errors" />
          <article>
            <h1>My Top Stories</h1>
            <div className="entry">
              {
                entries.slice(0, 3).map((entry) => <EntryList key={entry.id} {...entry} {...this.props} />)
              }
            </div>
          </article>
        </section>
      </div>
    );
  }
}

const mapStateToProps = ({
  profileReducer: { status, gottenProfile },
  updateProfileImageReducer: { statuss, updatedImage },
  entries: { entries, entryLoading },
}) => ({
  status,
  statuss,
  gottenProfile,
  updatedImage,
  entries,
  entryLoading,
});

export default connect(mapStateToProps, {
  getUserProfile: () => getProfile(),
  getAllEntries: () => getEntries(),
  updateImage: (data) => updateProfileImage(data),
})(Profile);
