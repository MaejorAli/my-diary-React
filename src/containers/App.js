import React from 'react';
import { BrowserRouter, Route } from 'react-router-dom';
import LandingPage from './LandingPage';
import '../styles/styles.scss';
import Signin from './Signin';

const App = () => (
  <BrowserRouter>
    <div>
      <Route exact path="/" component={LandingPage} />
      <Route exact path="/auth/signin" component={Signin} />
    </div>
  </BrowserRouter>
);

export default App;
