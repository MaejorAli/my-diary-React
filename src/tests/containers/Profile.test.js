import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Profile } from '../../containers/Profile';

const props = {
  signupUser: jest.fn(),
  signin: jest.fn(),
  PublishEntry: jest.fn(),
  getAllEntries: jest.fn(),
  GetEntries: jest.fn(),
  entries: [
    { id: 1 }, { id: 2 }, { id: 3 }, { id: 4 },
  ],
  entry: {},
  match: {
    params: { id: 1 },
  },
  deleteEntry: jest.fn(),
  deleteEntryReducer: jest.fn(),
  history: { push: jest.fn() },
  gottenProfile: {},
  getUserProfile: jest.fn(),
  status: 'true',
  statuss: 'true',
  updatedImage: 'true',
  entryLoading: '',
  updateImage: jest.fn(),
};

const wrapped = shallow(<Profile {...props} />);

describe('Profile container', () => {
  describe('render features', () => {
    test('view should render as expected', () => {
      const tree = toJson(wrapped);
      expect(tree).toMatchSnapshot();
    });
  });
  describe('Profile functionality', () => {
    it('should handle submit', () => {
      const submitButton = wrapped.find('#file-input');

      const mockedEvent = {
        target: {
          files: [
            {}, {},
          ],
        },
      };
      submitButton.simulate('change', mockedEvent);
    });
  });
});
