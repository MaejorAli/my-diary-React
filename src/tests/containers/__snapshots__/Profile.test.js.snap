// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile container render features view should render as expected 1`] = `
<div
  className="profile-body"
>
  <header>
    <div
      id="menu"
    >
      <Link
        className="menuitem"
        replace={false}
        to="/dashboard"
      >
        Home
      </Link>
      <Link
        className="menuitem"
        replace={false}
        to="/entries"
      >
        Entries
      </Link>
      <Link
        className="menuitem"
        replace={false}
        to="/entries/create"
      >
        New Entry
      </Link>
      <Link
        className="menuitem"
        replace={false}
        to="/signout"
      >
        Sign Out
      </Link>
    </div>
  </header>
  <div
    className="user-profile"
  >
    <img
      alt="Ali"
      className="avatar"
      id="image"
      src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mNk+P+/HgAFhAJ/wlseKgAAAABJRU5ErkJggg=="
    />
    <div
      className="username"
      id="username"
    >
       
    </div>
    <div
      className="email"
      id="email"
    />
    <h4>
      Published Stories 
      <span
        id="entryCount"
      />
      4
    </h4>
  </div>
  <section>
    <div
      className="part"
    >
      <div
        className="image-upload"
      >
        <label
          htmlFor="file-input"
        >
          <h5>
            Change Profile Picture
          </h5>
        </label>
        <input
          id="file-input"
          onChange={[Function]}
          type="file"
        />
      </div>
    </div>
    <div
      id="errors"
    />
    <article>
      <h1>
        My Top Stories
      </h1>
      <div
        className="entry"
      >
        <EntryList
          GetEntries={[MockFunction]}
          PublishEntry={[MockFunction]}
          deleteEntry={[MockFunction]}
          deleteEntryReducer={[MockFunction]}
          entries={
            Array [
              Object {
                "id": 1,
              },
              Object {
                "id": 2,
              },
              Object {
                "id": 3,
              },
              Object {
                "id": 4,
              },
            ]
          }
          entry={Object {}}
          entryLoading=""
          getAllEntries={
            [MockFunction] {
              "calls": Array [
                Array [],
              ],
              "results": Array [
                Object {
                  "isThrow": false,
                  "value": undefined,
                },
              ],
            }
          }
          getUserProfile={
            [MockFunction] {
              "calls": Array [
                Array [],
              ],
              "results": Array [
                Object {
                  "isThrow": false,
                  "value": undefined,
                },
              ],
            }
          }
          gottenProfile={Object {}}
          history={
            Object {
              "push": [MockFunction],
            }
          }
          id={1}
          key="1"
          match={
            Object {
              "params": Object {
                "id": 1,
              },
            }
          }
          signin={[MockFunction]}
          signupUser={[MockFunction]}
          status="true"
          statuss="true"
          updateImage={[MockFunction]}
          updatedImage="true"
        />
        <EntryList
          GetEntries={[MockFunction]}
          PublishEntry={[MockFunction]}
          deleteEntry={[MockFunction]}
          deleteEntryReducer={[MockFunction]}
          entries={
            Array [
              Object {
                "id": 1,
              },
              Object {
                "id": 2,
              },
              Object {
                "id": 3,
              },
              Object {
                "id": 4,
              },
            ]
          }
          entry={Object {}}
          entryLoading=""
          getAllEntries={
            [MockFunction] {
              "calls": Array [
                Array [],
              ],
              "results": Array [
                Object {
                  "isThrow": false,
                  "value": undefined,
                },
              ],
            }
          }
          getUserProfile={
            [MockFunction] {
              "calls": Array [
                Array [],
              ],
              "results": Array [
                Object {
                  "isThrow": false,
                  "value": undefined,
                },
              ],
            }
          }
          gottenProfile={Object {}}
          history={
            Object {
              "push": [MockFunction],
            }
          }
          id={2}
          key="2"
          match={
            Object {
              "params": Object {
                "id": 1,
              },
            }
          }
          signin={[MockFunction]}
          signupUser={[MockFunction]}
          status="true"
          statuss="true"
          updateImage={[MockFunction]}
          updatedImage="true"
        />
        <EntryList
          GetEntries={[MockFunction]}
          PublishEntry={[MockFunction]}
          deleteEntry={[MockFunction]}
          deleteEntryReducer={[MockFunction]}
          entries={
            Array [
              Object {
                "id": 1,
              },
              Object {
                "id": 2,
              },
              Object {
                "id": 3,
              },
              Object {
                "id": 4,
              },
            ]
          }
          entry={Object {}}
          entryLoading=""
          getAllEntries={
            [MockFunction] {
              "calls": Array [
                Array [],
              ],
              "results": Array [
                Object {
                  "isThrow": false,
                  "value": undefined,
                },
              ],
            }
          }
          getUserProfile={
            [MockFunction] {
              "calls": Array [
                Array [],
              ],
              "results": Array [
                Object {
                  "isThrow": false,
                  "value": undefined,
                },
              ],
            }
          }
          gottenProfile={Object {}}
          history={
            Object {
              "push": [MockFunction],
            }
          }
          id={3}
          key="3"
          match={
            Object {
              "params": Object {
                "id": 1,
              },
            }
          }
          signin={[MockFunction]}
          signupUser={[MockFunction]}
          status="true"
          statuss="true"
          updateImage={[MockFunction]}
          updatedImage="true"
        />
      </div>
    </article>
  </section>
</div>
`;
